### PR TITLE
AMQP-830 Enable Hostname Verification by default

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-830

Tested with a stand-alone boot app.

### CODE REVIEW

**Comment by artembilan** [comment]
Here we have slightly different error:
```
org.springframework.amqp.rabbit.core.RabbitTemplateDirectReplyToContainerIntegrationPubCFTests > testReceiveAndReplyBlocking FAILED
    org.springframework.amqp.AmqpConnectException
        Caused by: com.rabbitmq.client.ShutdownSignalException
```
But it is still related to channel shutdown...

File: src/reference/asciidoc/whats-new.adoc [hunk 1]
```diff
@@ -179,6 +179,12 @@ Also the `ConnectionBlockedEvent` and `ConnectionUnblockedEvent` events are emit
 
 See <<connections>> for more information.
 
+===== Connection Factory Bean Changes
+
+The `RabbitConnectionFactoryBean` now provides an `enabaleHostnameVerification` property; set it to `true` to enable host name verification.
+Also `setHostnameVerifier` is provided for when using Java 6; see the connection factory javadocs for more information.
```
Comment (artembilan) [inline src/reference/asciidoc/whats-new.adoc]: There is no such a functionality in this version.
Also the commit message is misleading.

Will fix on merge...

Comment (garyrussell) [reply src/reference/asciidoc/whats-new.adoc]: Oops - yes; thanks.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/d64e7fa3993dac577c0973e0caf8c31d27ef5e44

**Pull request closed without merging**